scripts/lint.py

```diff
@@ -748,6 +749,23 @@ def lint_file(filepath: str, args: Any) -> int:
     return num_errors
 
 
+def lint_crate_docs(should_ignore: Callable[[Any], bool]) -> int:
+    """Make sure ARCHITECTURE.md talks about every single create we have."""
+
+    crates_dir = Path(__file__).parent.parent / "crates"
+    architecture_md_file = Path(__file__).parent.parent / "ARCHITECTURE.md"
+
+    architecture_md = architecture_md_file.read_text()
+
+    error_count = 0
+    for crate in crates_dir.iterdir():
+        if not should_ignore(crate) and crate.name not in architecture_md:
+            print(f"Missing documentation for crate {crate.name} in ARCHITECTURE.md")
```

Comment:
This omits the negated test, i.e. that `ARCHITECTURE.md` doesn't mention crates that no longer exist

Reply:
Done